apps/examples/kernel_sample/pthread.c

```diff
@@ -0,0 +1,75 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/***********************************************************************
+ * examples/kernel_sample/pthread.c
+ ***********************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/compiler.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <pthread.h>
+
+#include "kernel_sample.h"
+
+#define NLOOPS 100
+static pthread_t thread;
+
+static void *empty_func(void *parameter)
+{
+	return NULL;
+}
+
+void pthread_test(void)
+{
+	int nloops = NLOOPS;
+	pthread_attr_t attr;
+	int status;
+
+	/* Save the initial memory usage before running the test. */
+
+#ifndef CONFIG_DISABLE_SIGNALS
+	struct mallinfo mem_before;
+
+#ifdef CONFIG_CAN_PASS_STRUCTS
+	mem_before = mallinfo();
+#else
+	(void)mallinfo(&mem_before);
+#endif
+	set_mmprevious(&mem_before);
```

Comment:
This looks unnecessary because of `check_test_memory_usages`.
It saves previous state.